src/InputFilter.js

```diff
@@ -41,13 +41,19 @@ export default function inputFilterFactory(store) {
 
     componentDidMount() {
       updateValue(this.props.initialSearch, this.props.onChange, true);
+      this.unsubscribe = store.on(value => {
+        this.setState({ value });
+      });
+    }
+
+    componentWillUnmount() {
+      this.unsubscribe();
     }
 
     componentWillReceiveProps(nextProps) {
       this.updateValue = debounce(updateValue, nextProps.debounceTime);
       if (nextProps.initialSearch !== this.props.initialSearch) {
         updateValue(nextProps.initialSearch, this.props.onChange);
-        this.setState({ value: nextProps.initialSearch });
```

Comment:
@jdlehman why are you removing this line?

Reply:
I now set this state in the subscription to `changeInputValue`. Previously this component was the only one setting state so it did not need to subscribe, but now that it is getting changes externally I consolidated `value`'s `setState`: https://github.com/jdlehman/react-fuzzy-filter/pull/12/files#diff-56411d070709ec944811808575ee1688R45